src/Umbraco.Cms.Integrations.SEO.SemrushTools/Umbraco.Cms.Integrations.SEO.SemrushTools/Services/SemrushCachingService.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+
+using Newtonsoft.Json;
+
+using Umbraco.Cms.Integrations.SEO.SemrushTools.Models.Dtos;
+using Umbraco.Core.Cache;
+
+namespace Umbraco.Cms.Integrations.SEO.SemrushTools.Services
+{
+    public class SemrushCachingService: ISemrushCachingService<RelatedPhrasesDto>
+    {
+        private readonly IAppPolicyCache _cache;
+
+        public SemrushCachingService(AppCaches appCaches)
+        {
+            _cache = appCaches.RuntimeCache;
+        }
+
+        public bool TryGetCachedItem(out RelatedPhrasesDto item, string key)
+        {
+            var serializedItem = _cache.GetCacheItem<string>(key);
+
+            item = string.IsNullOrEmpty(serializedItem)
+                ? new RelatedPhrasesDto()
+                : JsonConvert.DeserializeObject<RelatedPhrasesDto>(serializedItem);
```

Comment:
We should be explicit about the serialization settings used here, as it could be that a developer redefines the global defaults elsewhere in their project.  I.e. define them statically somewhere and pass them in use the parameter available in `DeserializeObject` and `SerializeObject`.

Reply:
Item addressed.